### PR TITLE
[menu-bar] Bump electron to 33.2.0 and react to 18.3.1

### PR DESCRIPTION
**Pull request by gabrieldonadel**
# Why

Orbit dependencies are getting quite outdated

# How

- Upgrade all electron-related libraries and remove unnecessary dependencies 
- Fix logs not showing up on windows

# Test Plan

Run menubar locally on macOS

### CODE REVIEW

File: apps/menu-bar/src/modules/Storage.ts [hunk 1]
```diff
@@ -112,3 +112,22 @@ export function resetApolloStore() {
...
+    }
+  }
+
+  get(): Log[] {
+    if (Platform.OS === 'web') {
+      return JSON.parse(localStorage.getItem('logs') ?? '[]');
+    }
+    return this.logs ?? [];
+  }
+}
```
Comment (gabrieldonadel) [inline apps/menu-bar/src/modules/Storage.ts:133]: In order for the logs to work across multiple windows on react-native web we need to use some sort of storage solution like localStorage

File: apps/menu-bar/electron/src/main.ts [hunk 2]
```diff
@@ -51,6 +51,11 @@ const createMainWindow = () => {
     mainWindow.loadURL(`file://${path.join(__dirname, '../../../app/dist/index.html')}`);
   }
 
+  mainWindow.webContents.once('dom-ready', () => {
+    // Only keep the current session in the logs
+    mainWindow.webContents.executeJavaScript("localStorage.setItem('logs', '[]')");
+  });
```
Comment (gabrieldonadel) [inline apps/menu-bar/electron/src/main.ts:57]: We need to reset localStorage logs on launch to ignore the logs from the previous session

File: apps/menu-bar/package.json [hunk 3]
```diff
@@ -19,25 +19,27 @@
...
-    "react": "18.2.0",
-    "react-dom": "18.2.0",
+    "react": "18.3.1",
+    "react-dom": "18.3.1",
     "react-native": "0.76.1",
     "react-native-macos": "0.76.0",
     "react-native-mmkv": "^2.10.2",
     "react-native-svg": "15.8.0",
     "react-native-url-polyfill": "^2.0.0",
-    "react-native-vector-icons": "^9.2.0",
```
Comment (gabrieldonadel) [inline apps/menu-bar/package.json:40]: no longer used

File: apps/menu-bar/package.json [hunk 4]
```diff
@@ -19,25 +19,27 @@
...
-    "@expo/metro-runtime": "~3.2.3",
+    "@expo/metro-runtime": "~4.0.0",
     "@expo/styleguide-native": "^1.0.1",
-    "@fluentui/react-components": "^9.46.4",
+    "@fluentui/react-checkbox": "^9.2.40",
     "@fluentui/react-icons": "^2.0.227",
+    "@fluentui/react-progress": "^9.1.90",
+    "@fluentui/react-provider": "^9.18.0",
+    "@fluentui/react-switch": "^9.1.97",
+    "@fluentui/react-theme": "^9.1.22",
```
Comment (gabrieldonadel) [inline apps/menu-bar/package.json:29]: latest release of `@fluentui/react-components` is currently broken so we need to use the individual packages directly

**Review by alanjhughes — APPROVED** [review]

**Pull request merged**